Python/emscripten_signal.c

```diff
@@ -38,19 +38,17 @@ _Py_CheckEmscriptenSignals(void)
     }
 }
 
-
 #define PY_EMSCRIPTEN_SIGNAL_INTERVAL 50
 static int emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
 
 void
 _Py_CheckEmscriptenSignalsPeriodically(void)
 {
-    if (!Py_EMSCRIPTEN_SIGNAL_HANDLING) {
-        return;
-    }
-    emscripten_signal_clock--;
     if (emscripten_signal_clock == 0) {
         emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
         _Py_CheckEmscriptenSignals();
     }
+    else {
+        emscripten_signal_clock--;
+    }
```

Comment:
Why is this being changed? It seems fine as-is (and it's not clear why the `Py_EMSCRIPTEN_SIGNAL_HANDLING` bit is being removed)...

Reply:
`Py_EMSCRIPTEN_SIGNAL_HANDLING` is tested in `_Py_CheckEmscriptenSignals()`, so we don't need to check it twice.

Removing it here and decrementing `emscripten_signal_clock` only when `emscripten_signal_clock != 0` means that the `DOPT_IF` in `RESUME_QUICK` only needs to check `emscripten_signal_clock` and not `Py_EMSCRIPTEN_SIGNAL_HANDLING`.

@hoodmane can you confirm that this change is correct?

Reply:
It won't lead to any incorrect behavior. The decrement change makes it so you have to check for `emscripten_signal_clock == 0` rather than checking it for `1` which is probably more understandable. But I don't see what the point of removing the guard is: if we keep the guard then when `Py_EMSCRIPTEN_SIGNAL_HANDLING` is `false`, the value of `emscripten_signal_clock` never changes and is never zero so we never trigger the `DOPT_IF`. This seems like it would better satisfy the goal of not paying a performance cost when the signal handling is off.

Reply:
That makes sense. 
We keep `emscripten_signal_clock == 0` as the trigger, but we only decrement if `Py_EMSCRIPTEN_SIGNAL_HANDLING`

Reply:
We need to escape the condition `emscripten_signal_clock == 0 && Py_EMSCRIPTEN_SIGNAL_HANDLING`. If we test `Py_EMSCRIPTEN_SIGNAL_HANDLING` first, then we will constantly hit the deopt case.